File: Xaas/Python_이상탐지 알고리즘/kmeanlbg.py
import numpy as np
from k_means import k_means

def kmeanlbg(d, k):
    nc = d.shape[1]
    
    x, esq, j = k_means(d, 1)

    m = 1
    iter = 0
    while m < k:
        iter += 1
        n = min(m, k - m)
        m = m + n
        
        e = 1e-4 * np.sqrt(esq) * np.random.rand(1, nc)
        
        x, esq, j = k_means(d, m, np.vstack([x[:n, :] + e[:n, :], x[:n, :] - e[:n, :], x[n:m-n, :]]))
        
    return x, esq, j

File: Xaas/Comparative_study/0_PCA_SS/kmeanlbg.m
function [x,esq,j] = kmeanlbg(d,k)
%KMEANLBG Linde-Buzo-Gray

nc=size(d,2);
[x,esq,j]=k_means(d,1);
m=1;
iter = 0;
while m<k
   iter = iter + 1;
   n=min(m,k-m);
   m=m+n;
   e=1e-4*sqrt(esq)*rand(1,nc);
   [x,esq,j]=k_means(d,m,[x(1:n,:)+e(ones(n,1),:); x(1:n,:)-e(ones(n,1),:); x(n+1:m-n,:)]);

end
